packages/core/admin/ee/admin/src/pages/SettingsPage/pages/SingleSignOn/hooks/useSettingsForm.ts

```diff
@@ -0,0 +1,244 @@
+import * as React from 'react';
+import produce from 'immer';
+import pick from 'lodash/pick';
+import set from 'lodash/set';
+import { ValidationError } from 'yup';
+
+import {
+  getYupInnerErrors,
+  useFetchClient,
+  useNotification,
+  useOverlayBlocker,
+} from '@strapi/helper-plugin';
+import omit from 'lodash/omit';
+
+import { formatAPIErrors } from '../../../../../../../../admin/src/utils/formatAPIErrors';
+
+interface Action {
+  type: 'GET_DATA_SUCCEEDED' | 'ON_CANCEL' | 'ON_CHANGE' | 'ON_SUBMIT' | 'ON_SUBMIT_SUCCEEDED' | 'SET_ERRORS';
+  data?: {
+    [key: string]: unknown;
+  };
+  errors?: Record<string, unknown>;
+  fieldsToPick?: string[];
+  keys?: string;
+  value?: unknown;
+}
```

Comment:
This isn't a good way to type Action, what you should do is define every action separately and then join them as a union, they're then discriminated by their type and you know what things to expect. IMO i would change it because it goes against how other reducers are typed, but i'm not gonna block the PR for it.

Reply:
it'd also probably clean up a lot of the code you had to write to say "if it's not there do this"